pageserver/src/tenant/timeline/detach_ancestor.rs

```diff
@@ -41,6 +41,26 @@ pub(crate) enum Error {
     Unexpected(#[source] anyhow::Error),
 }
 
+impl From<Error> for ApiError {
+    fn from(value: Error) -> Self {
+        match value {
+            e @ Error::NoAncestor => ApiError::Conflict(e.to_string()),
+            // TODO: ApiError converts the anyhow using debug formatting ... just stop using ApiError?
+            e @ Error::TooManyAncestors => ApiError::BadRequest(anyhow::anyhow!("{}", e)),
+            e @ Error::ShuttingDown => ApiError::InternalServerError(e.into()),
```

Comment:
ApiError has a shutdown variant which maps to 503 -- if possible we should avoid returning 500 intentionally

Reply:
And I did the 500 thinking because there is 1001 cases where a shutdown error is not properly mapped (because of anyhow usage).

Reply:
>Ok, I had documented that 500 is for whatever transient. But yeah, why not it could be a 503. Just feels like 503 is given if something cannot be started.

I think of 503 as the "gaps" in a service -- it covers the time from when we start shutting down, to the time when we become available for service again.  500 should be used for genuinely unexpected things (bugs), or cases that we can't distinguish from bugs (in this case, because we have anyhow::Errors way further down the stack).